### DISCUSSION TITLE
how to have 2 different UIs for Index (/) page & getServerSideProps() call to??

### DISCUSSION THREAD

**Discussion by dhavalveera** [vercel/next.js#39636] [Help]
Hello,

https://www.swiggy.com/ something like this in Next.js, as on the Index/Home (route: '/') Page first they're asking for the location, and once the location was passed they're showing the data on the same Index/Home (route: '/') Page, and here can I able to use getServerSideProps() once I get the location for the same Index/Home (route: '/') Page??

anyone can please help me

**Comment by timfuhrmann** [comment]
You could create a cookie holding the user's location and check said cookie's presence using Next.js' middleware - if present you rewrite `/` to actually be e.g. `/search`. 

On `/search` you could pass that cookie's data to the client using `getServerSideProps` or read from the client directly.
